src/sec_adapter_soc_provisioning.h

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright 2025 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef SEC_ADAPTER_SOC_PROVISONING_H
+#define SEC_ADAPTER_SOC_PROVISONING_H
+#include "sa.h"
+#include "sec_adapter_cipher.h"
+#include "sec_adapter_key_legacy.h"
+#include "sec_adapter_processor.h"
+#include "sec_security_utils.h"
+#include <memory.h>
+
+
+// Parameters to pass in the key provision 
```

Comment:
These seem to be far removed from where the array and structure is that declares the objects. If it is not possible to group these with the object(s) that they are associated with, you should add a comment to reference where the rest of the object is declared so these are easily updated in the fututre.